v2.2/architecture/transaction-layer.md

```diff
@@ -102,21 +102,30 @@ After setting up this bookkeeping, the request is passed to the `DistSender` in
 
 ### Transaction records
 
-When a transaction starts, `TxnCoordSender` writes a transaction record to the range containing the first key modified in the transaction. As mentioned above, the transaction record provides the system with a source of truth about the status of a transaction.
+To track the status of a transaction's execution, we write a value called a transaction record to our key-value store. This lets any transaction check the status of any other transaction, which is important for handling concurrency.
 
-The transaction record expresses one of the following dispositions of a transaction:
+When a `TxnCoordSender` begins transaction, it determines where to write the transaction record (this location will always be the same range as the first key in the transaction), and then includes a reference to the transaction record in all of its write intents.
 
-- `PENDING`: The initial status of all values, indicating that the write intent's transaction is still in progress.
-- `COMMITTED`: Once a transaction has completed, this status indicates that the value can be read.
-- `ABORTED`: If a transaction fails or is aborted by the client, it's moved into this state.
+However, the transaction record itself isn't created until one of the following conditions occur:
 
-The transaction record for a committed transaction remains until all its write intents are converted to MVCC values. For an aborted transaction, the transaction record can be deleted at any time, which also means that CockroachDB treats missing transaction records as if they belong to aborted transactions.
+- The write operation commits
+- The `TxnCoordSender` heartbeats the transaction
+- An operation forces the transaction to abort
+
+Given this mechanism, the transaction record uses the following states:
+
+- `PENDING`: Indicates that the write intent's transaction is still in progress.
+- `COMMITTED`: Once a transaction has completed, this status indicates that write intents can be treated as committed values.
+- `ABORTED`: Indicates that the transaction was aborted and its values should be discarded.
+- _Record does not exist_: Because records might not have been written yet, a missing record require the contending transaction to look at the write intent's timestamp. If it was created within the last two seconds, the transaction is treated as if it's `PENDING`, otherwise the transaction is considered `ABORTED`.
```

Comment:
We might want to specify what this write intent is. Probably a condensed version of "the write intent that the contending transaction observed which drove it to look for the transaction record."

Reply:
Tried to fix this up with a better explanation and better pronouns. Can I get you to take a quick look?